text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,502 @@
+
+* Implementation is extensible for future features such as:
+  * Self-managed Stack Sets
+  * Stack Set Instance Add/Remove
+  * Additional asset management
+
+# Design
+## Design Overview:
+* [1] [CLI Integration - Stack/Stack Set Collection](#1-cli-integration---stackstack-set-collection)
+  * CLI user experience and how stack/stack sets will be collected under the hood
+* [2] [Construct Changes (CDK app) - Stack Set Artifact Type Synthesis](#2-construct-changes-cdk-app---stack-set-artifact-type-synthesis)
+  * How stack sets will be defined and prepared for CLI consumption
+    * New Stack Set core construct type and artifact type
+    * New synthesizer for Stack Sets
+* [3] [CLI Changes - Deployment of Stack Set](#3-cli-changes---deployment-of-stack-set)
+  * How the CLI will orchestrate the deployment of stack sets
+    * Primitives added for making API calls to create/update stack sets
+    * Branching between deploying stacks vs stack sets
+* [4] [File Assets Management](#4-file-assets-management)
+  * How file assets are defined
+  * How bootstrapping works with stack sets
+  * How file assets are made available to accounts “unmanaged” by CDK and in regions that the stack set deploys to
+
+## [1] CLI Integration - Stack/Stack Set Collection
+## High Level CLI/Deploy Flow
+![Architecture Diagram](../images/self-managed-stack-sets/high_level_cli_deploy_flow.jpg)
+1. CLI calls list/synth/deploy
+1. Subprocess invokes CDK App
+    1. StackSet construct generates template and artifact is added to manifest as a new artifact type
+1. CLI reads manifest
+1. CLI deploys to AWS environment
+
+### Supported Commands:
+* cdk diff
+* cdk synth
+* cdk deploy
+* cdk list
+
+### [1.1] Commands/flags for stack sets
+At a high-level, the flow for stacks vs stack sets are nearly identical with a few different configuration parameters and end deployment logic.
+
+**Option 1**: Commands will remain the same and perform actions for both stack/stack sets **(Preferred)**
+
+In this option, the commands can be ran as the are today, and under the hood logic will be branched based on artifact type.
+
+* **Pros**:
+  * Unified customer experience for deploying infrastructure
+  * Allows deployment of stacks/stack sets in the same execution
+* **Cons**:
+  * Risk of breaking current stacks flow
+  * Under the hood, there will need to be some branching logic between types and changes to stacks collection since the current code is tightly tied to stack artifact types
+
+**Option 2:** Add Separate commands or flag to distinguish between stacks/stack sets
+
+* **Pros**:
+  * Isolates stack sets to its own flows -- less risk of breaking stacks flow
+* **Cons**:
+  * Odd user experience since the user will need to think about when to use commands for stack vs stack sets
+  * Some code duplication, since the stack set flow is nearly identical, with some differences in input parameters
+  * Commands will need to be ran twice if the CDK app contains both stacks/stack sets
```

Comment:
Let's go with Option 1, and remove Option 2.

Reply:
Remove -- yeah added originally to have an opposing alternative